funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_worker.py

```diff
@@ -124,29 +122,44 @@ def start(self):
 
     def execute_task(self, task_id: str, task_body: bytes) -> dict:
         log.debug("executing task task_id='%s'", task_id)
-        exec_start_ms = _now_ms()
+        exec_start = TaskTransition(
+            timestamp=time.time_ns(), state=TaskState.EXEC_START, actor=ActorName.WORKER
+        )
 
         try:
             result = self.call_user_function(task_body)
         except Exception:
             log.exception("Caught an exception while executing user function")
+            exec_end = TaskTransition(
+                timestamp=time.time_ns(),
+                state=TaskState.EXEC_END,
+                actor=ActorName.WORKER,
+            )
             result_message = dict(
                 task_id=task_id,
                 exception=get_error_string(),
                 error_details=get_result_error_details(),
-                exec_start_ms=exec_start_ms,
-                exec_end_ms=_now_ms(),
+                task_statuses=[exec_start, exec_end],
```

Comment:
(Just a comment)

Moving to a concept of `task_statuses` as a list (rather than hard-coded set of data points) makes it more stark in my mind that the "result" is really just the task, repackaged.  That is, why are we setting the `task_statuses` field anew here, rather than tacking on to the previous set of tasks?  Seems awry, yeah?